src/nilql.ts

```diff
@@ -722,19 +721,39 @@ async function hkdfDerive(
   length: number,
 ): Promise<Uint8Array> {
   // Step 1: Extract
-  const prk = createHmac("sha512", salt || Buffer.alloc(0))
-    .update(ikm)
-    .digest();
+  const defaultSalt = new Uint8Array([0x00]); // Single zero byte as default salt
+  const hmacSalt = salt || defaultSalt;
+
+  const key = await crypto.subtle.importKey(
```

Comment:
If targeting newer environments (which we are), I recommend using `globalThis` when accessing global objects that aren't explicitly imported. `globalThis` works consistently across different JS runtimes (browsers, node, web workers) and is meant to replace environment-specific globals like `window.crypto`.

You can also add and centralize validation to handle environments where it might not be available:

```ts
function getCrypto(): Crypto {
  if (!globalThis.crypto) {
     throw new Error("Crypto web-api not available but required")
  }
  return globalThis.crypto
}
```

To make the Crypto type available we might need to tweak tsconfig.json ... not sure atm